lib/file_store/s3_store.rb

```diff
@@ -104,10 +104,15 @@ def path_for(upload)
       FileStore::LocalStore.new.path_for(upload) if url && url[/^\/[^\/]/]
     end
 
-    def url_for(upload)
-      if upload.private?
+    def url_for(upload, params = {})
```

Comment:
I am not a fan of passing the web parameters directly through to service objects / models. In this case it seems the `s3_store` class is even comparing against a string.

It would be better if we added a keyword argument here: `def url_for(upload, force_download: false)` and then have the controller set that value to `params[:dl] == "1"`.

Reply:
Fixed. I initially opted for passing the parameters to avoid nesting `if` statements in the controller.

Reply:
You could avoid it in this case with `force_downloaded: params[:dl] == "1"` though?